validator/sawtooth_validator/state/client_handlers.py

```diff
@@ -32,182 +32,222 @@
 LOGGER = logging.getLogger(__name__)
 
 
-class StateCurrentRequestHandler(Handler):
-    def __init__(self, current_root_func):
-        self._current_root_func = current_root_func
+class _ClientHandler(Handler):
+    """
+    Base Client Handler, simplifies logic branching by creating a
+    result with a response body that can only be set once
+
+    Args:
+        request_proto - protobuf class for the request
+        response_proto - protobuf class for the response
+        result_type - the Message enum for the response
+    """
+    def __init__(self, request_proto, response_proto, result_type):
+        self._request = request_proto()
+        self._response_proto = response_proto
+        self._result_type = result_type
+        self._result = None
```

Comment:
That I don't know, probably a question for Shawn/Andi/Boyd? For what it's worth, the db and blockstore were already getting saved statefully to the handlers previous to this refactor. I think the only new thing that might create an issue is the request proto class, which gets instantiated during `init`. That could get moved to `handle` if someone who knows more than me about python and multithreading thinks it could cause issues.